### ISSUE TITLE
clang-tidy does not recognize llvm::seq idom.

### ISSUE THREAD

**Issue by Meinersbur** [google/llvm-premerge-checks#265]
**Describe the bug**
```
for (auto i : llvm::seq<int>(0, N))
```
is an alternative for 
```
for (int i = 0; i < N; ++i)
```
But the pre-merge check complains with:
> clang-tidy: warning: invalid case style for variable 'i' [readability-identifier-naming]

Encountered at the pre-merge check of https://reviews.llvm.org/D92974.

See also the discussion on the [mailing list](https://lists.llvm.org/pipermail/llvm-dev/2019-June/132895.html), [unit tests](https://github.com/llvm/llvm-project/blob/fd78012b9e592d1d5e1904cf2dbfdb1a17b803a4/llvm/unittests/ADT/SequenceTest.cpp#L20), and the coding standard document about [range-based for-loops](https://llvm.org/docs/CodingStandards.html#use-range-based-for-loops-wherever-possible), especially one of the [examples](https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements).

**Comment by metaflow** [comment]
Thank you for the report @Meinersbur!
clang-tidy settings for /llvm https://github.com/llvm/llvm-project/blob/main/llvm/.clang-tidy tells that variables should be CamelCase and it rightfully complains about that.
It does not seems that issue is with `llvm:seq`.

**Comment by Meinersbur** [comment]
However, somehow it does not flag the equivalent `for (int i = 0; i < N; ++i)` to be changed to `for (int I = 0; I < N; ++I)`. I interpret the LLVM coding standard that both are allowed, and is common in the current code base. Sweeping over the clang-tidy code base, I could not find where this is implemented.

Reading from you response, you think this is a problem with how LLVM repository itself configured clang-tidy?

**Comment by metaflow** [comment]
Are you sure about that? I have just added `for (int i = 0; i < 10; i++);` and got
```
/root/llvm-project/llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp:1252:12: warning: invalid case style for variable 'i' [readability-identifier-naming]
  for (int i = 0; i < 10; i++);
           ^      ~       ~
           I      I       I
```

Note that for premerge checks we only run clang-tidy-diff over new changes. So existing warnings will not be shown (if you see lowercase example in the same file).

**Comment by metaflow** [comment]
@Meinersbur closing this issue. As I see it tools operated correctly.

**Comment by Meinersbur** [comment]
np. Using lowercase for an integer iterator is common throughout the codebase though, but it seems I have never seen a clang-tidy warning about it, including https://reviews.llvm.org/D92974 where I changed llvm::seq to the C-style idiom. But that did not get to the clang-tidy stage because the patch does not apply cleanly (anymore).
